Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -69,8 +69,80 @@ struct _WebKitUserContentManagerPrivate {
 
 WEBKIT_DEFINE_TYPE(WebKitUserContentManager, webkit_user_content_manager, G_TYPE_OBJECT)
 
+/**
+ * WebKitUserScriptMessage: (ref-func webkit_user_script_message_ref) (unref-func webkit_user_script_message_unref)
+ *
+ * A helper class that sends user provided script to the signal handler.
+ *
+ * Since: 2.38
+ */
+struct _WebKitUserScriptMessage {
+    _WebKitUserScriptMessage(WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler)
+        : completionHandler(completionHandler)
+        , referenceCount(1)
+    {
+        JSCContext* jscContext = jsc_context_new();
+        JSValueRef jsValueRef = serializedScriptValue.deserialize(jscContextGetJSContext(jscContext), nullptr);
+        message = jscValueCreate(jscContext, jsValueRef);
```

Comment:
Here we should use the internal utility functions we have to create or reuse a shared context used for deserializing JS values. You can see an example in `WebKitJavascriptResult.cpp`. It would be like this:

```suggestion
        auto* jsContext = SharedJavascriptContext::singleton().getOrCreateContext();
        message = jscContextGetOrCreateValue(jsContext, serializedScriptValue.deserialize(jscContextGetJSContext(jscContext), nullptr));
```

The important bit is using the `getOrCreateContext()` method: it's somewhat expensive to create new JS execution contexts, so here using the shared really helps. The rest is the same you had&mdash;it looks a bit nicer to me as a single line, but there was nothing wrong with your approach.